kiwi/system/prepare.py

```diff
@@ -404,19 +407,20 @@ def pinch_system(
                 )
             )
             try:
-                if manager is None:
-                    package_manager = self.xml_state.get_package_manager()
-                    release_version = self.xml_state.get_release_version()
-                    manager = PackageManager.new(
-                        repository=Repository.new(
-                            self.root_bind, package_manager
-                        ),
-                        package_manager_name=package_manager,
-                        release_version=release_version
+                with ExitStack() as stack:
+                    if manager is None:
+                        package_manager = self.xml_state.get_package_manager()
+                        release_version = self.xml_state.get_release_version()
+                        repo = Repository.new(self.root_bind, package_manager)
+                        stack.push(repo)
+                        manager = PackageManager.new(
+                            repository=repo,
+                            package_manager_name=package_manager,
+                            release_version=release_version
+                        )
```

Comment:
How is this branch reachable? The type hint says that `manager` is an instance of `PackageManagerBase` not `Optional[PackageManagerBase]`

Reply:
The type hint is wrong. pinch_system is allowed to be called without a pre-created PackageManager instance. I will fix it